### PR TITLE
build: add GitHub Actions workflow for discovery sync

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This is in pursuit of enabling default branch protection.  This PR does a few things:
- It introduces a `npm run update-discovery` command, which updates all discovery files locally 
- ~~It runs said script, which causes a lot of changes (which I want feedback on @vchudnov-g).  Some of the changes are sorting keys to reduce diffs in future commits.~~
- It adds an automated workflow for running this process nightly
- It adds an automated workflow for auto-approving and merging the PR

After something like this lands and is verified, the bot that auto-pushes to master would no longer be needed.  We could also enable default branch protection rules and codeowners.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "private": "true",
+  "version": "1.0.0",
+  "description": "The Discovery Artifact Manager is intended to facilitate testing, publishing, and synchronization of generators and artifacts for client libraries and generated code samples of Google APIs defined by the API Discovery Service.",
+  "main": "update-discovery.js",
+  "scripts": {
+    "update-discovery": "node update-discovery.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/JustinBeckwith/discovery-artifact-manager.git"
```
Comment (chingor13) [inline package.json:12]: googleapis/discovery-artifact-manager?

File: .github/workflows/update-discovery.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Update Discovery
+
+on:
+  schedule:
+    - cron: "0 1 * * *"
```
Comment (chingor13) [inline .github/workflows/update-discovery.yml:5]: I think the current executor happens every 3 hours. A few of our autosynth jobs run around 1 or 2 AM PST as well.

File: .github/workflows/update-discovery.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        run: |
+          git config --global user.email "yoshi-automation@google.com"
+          git config --global user.name "Yoshi Automation Bot"
+          git checkout -b action/auto-update-discovery
+          git commit -a -m "chore: automatic update of discovery json"
+          git push -f origin action/auto-update-discovery
+      - name: Open PR
+        uses: actions/github-script@v2
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (chingor13) [inline .github/workflows/update-discovery.yml:29]: The default github-actions cannot sign the CLA.

```suggestion
          github-token: ${{ secrets.YOSHI_CODE_BOT_TOKEN }}
```

**Review by chingor13 — CHANGES_REQUESTED** [review]
Maybe let the first opened PR make the changes?

**Comment by chingor13** [comment]
Can we compile and run the existing go code that does the fetching and (refactor to node|implement sorting on the go version) in a subsequent PR?

**Comment by JustinBeckwith** [comment]
@chingor13 I think I got everything here

**Comment by chingor13** [comment]
[App Engine Task](https://github.com/googleapis/discovery-artifact-manager/blob/master/server/tasks/discovery_artifact_manager.py#L87-L96) which shells out to go.

[Go update binary](https://github.com/googleapis/discovery-artifact-manager/blob/master/src/main/updatedisco/main.go) 

[Go business logic](https://github.com/googleapis/discovery-artifact-manager/blob/master/src/main/common/updatedisco.go)

File: update-discovery.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Google LLC
```
Comment (SurferJeffAtGoogle) [inline update-discovery.js:1]: This 100 lines of new javascript is replacing about 100 lines of python that has been running for years.  

Please add tests that confirm this new code generates the same output for the same input as the old code.  Fuzzed tests would be ideal, but I won't insist.

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]
If we merge this PR, the bot will still be needed because there are other endpoints on the bot triggered by other cron jobs.

What should we do with https://github.com/googleapis/discovery-artifact-manager/pull/158?

**Comment by JustinBeckwith** [comment]
So I've come to the conclusion that this was a bad idea 😆  I was messing around with the final list of ~5 non-branch protected repos, and should have looked closer to see that @SurferJeffAtGoogle already did this in a much, much better way in #158.  Apologies for the confusion folks!

**Pull request closed without merging**